zoo/checklists/views.py

```diff
@@ -0,0 +1,124 @@
+import attr
+from django import forms
+from django.core.exceptions import SuspiciousOperation
+from django.http import Http404, JsonResponse
+from django.views.generic import ListView, TemplateView
+
+from . import models
+from ..services import models as service_models
+from .steps import STEPS
+
+
+class GlobalChecklistsView(ListView):
+    model = service_models.Service
+    template_name = "checklists/global_checklists.html"
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(status=service_models.Status.BETA.value)
+            .prefetch_related("checkmarks")
+        )
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["service_list"] = [
+            {
+                "name": service.name,
+                "owner_slug": service.owner_slug,
+                "name_slug": service.name_slug,
+                "completed_checklist_steps": service.checkmarks.count,
+                "total_checklist_steps": sum(
+                    [len(steps) for tag, steps in STEPS.items() if tag in service.tags]
```

Comment:
Maybe it would make sense to create a new function/method or property in the `Service` class which would return only the relevant steps. So that you just have something like:

`[len(steps) for tag, steps in service.steps.items()]`

Since you do similar thing here: https://github.com/kiwicom/the-zoo/blob/f312b29c0a5a700d43f1870df648e3fc8bdbcd71/zoo/checklists/views.py#L72-L73

Reply:
I won't add it to the model as that means adding a dependency that is not really needed for such small thing. But I agree that maybe a function could be good to have, even if is a one-line function.